src/Engine/Spells/CastSpellInfo.cpp

```diff
@@ -2143,13 +2036,10 @@ void CastSpellInfoHelpers::castSpell() {
                         }
 
                         initSpellSprite(&pSpellSprite, spell_level, spell_mastery, pCastSpell);
-                        pSpellSprite.vPosition.x = pActors[monster_id].vPosition.x;
-                        pSpellSprite.vPosition.y = pActors[monster_id].vPosition.y;
-                        pSpellSprite.vPosition.z = pActors[monster_id].uActorHeight;
+                        pSpellSprite.vPosition = Vec3i(pActors[monster_id].vPosition.x, pActors[monster_id].vPosition.y, pActors[monster_id].uActorHeight);
```

Comment:
Honestly, this looks like a bug. Using `pActors[monster_id].uActorHeight` for z coord. Let's add a TODO.